content/recipes/typegoose.md

```diff
@@ -0,0 +1,224 @@
+### Typegoose (Mongoose)
+
+[Typegoose](https://github.com/typegoose/typegoose) is a well maintained TypeScript wrapper for the [Mongoose ODM](https://mongoosejs.com/) library and is similar in nature to [the Mongoose Module](/techniques/mongodb) Nest provides. However, Typegoose's TypeScript support is more expansive, making your entity/model classes and your work with Mongoose, and in turn MongoDB, more expressive. Typegoose also offers extras like a specialized logger, a few extra types for a much better and more concise usage of Typegoose (and Mongoose), some added type guard methods, a number of specialized functions and more.  
+
+This recipe will get you started working with Typegoose in NestJS.
+
+#### Getting started
+
+To begin, you'll need to install the following dependencies into your [already created NestJS app](/first-steps):
+
+```typescript
+$ npm install --save mongoose @typegoose/typegoose @m8a/nestjs-typegoose
```

Comment:
Are we sure we want to link a library that has ~500 weekly downloads in the docs 🤔 ?

Reply:
@kamilmysliwiec - It's a chicken and egg dilemma. :grin: 

The @typegoose/typegoose package has 79k downloads a week. That's the popularity we'd want to possibly hook into. And vice-versa of course. 

Typegoose is a very good alternative for using Mongoose. The Typegoose package is very well maintained and documented and I take care of the nestjs-typegoose module updates now too. 

If more people see the recipe, the more the module will be downloaded. :wink: 

We've also been promoting Typegoose (mostly me) in the Discord server and the database forum on Discord has had the tag "mongoose/ typegoose" for several months now. 

And, if I may be so bold, the Nest team could just take over the nestjs-typegoose module and drop the Mongoose module altogether (sunset it). It would be a lot less effort from a maintenance standpoint and the Typegoose docs are much more detailed too.  :grinning:  Just an idea.....

Scott

Reply:
Theoretically this recipe could replace the Mongoose recipe too. It's outdated. 

https://docs.nestjs.com/recipes/mongodb

Scott

Reply:
I'd also like to put in that I've been helping in the forum a lot and I'm also willing to help continue to support [this package](https://www.npmjs.com/package/@m8a/nestjs-typegoose), if it became THE official _suggested_ module for using Mongoose with Nest. That means, I'd suggest even taking down the Nest mongoose module. That would save you all time and effort, by putting the support on the community and get Nest users something they can be very successful with using Mongoose with Nest!

Scott